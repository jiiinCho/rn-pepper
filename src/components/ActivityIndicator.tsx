import * as React from 'react';
import { Animated, Easing, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';

import { useInternalTheme } from '../core';
import type { ThemeProp } from '../types';

const DURATION = 2400;

const DefaultIndicatorSize = {
  small: 24,
  large: 48,
};

type LoaderSize = keyof typeof DefaultIndicatorSize;

type Props = React.ComponentPropsWithRef<typeof View> & {
  // Show indicator
  animating?: boolean;
  // Spinner color
  color?: string;
  // hide indicator when not animating
  hidesWhenStopped?: boolean;
  // Indicator size
  size?: LoaderSize | number;
  style?: StyleProp<ViewStyle>;
  theme?: ThemeProp;
};

/**
 * Activity indicator is used to present progress of some activity in the app.
 * It can be used as a drop-in for the ActivityIndicator shipped with React Native.
 */
const ActivityIndicator = ({
  animating = true,
  color: indicatorColor,
  hidesWhenStopped = true,
  size: indicatorSize = 'small',
  style,
  theme: themeOverrides,
  ...rest
}: Props) => {
  const theme = useInternalTheme(themeOverrides);
  const { current: timer } = React.useRef<Animated.Value>(new Animated.Value(0));
  const { current: fade } = React.useRef<Animated.Value>(
    new Animated.Value(!animating && hidesWhenStopped ? 0 : 1),
  );

  const rotation = React.useRef<Animated.CompositeAnimation | undefined>(undefined);

  const {
    animation: { scale },
  } = theme;

  const startRotation = React.useCallback(() => {
    // Show indicator
    // TODO: What is this for? toValue: 1 won't do anything because starting value is 1?
    Animated.timing(fade, {
      duration: 200 * scale,
      toValue: 1,
      isInteraction: false,
      useNativeDriver: true,
    }).start();

    // Circular animation in loop
    if (rotation.current) {
      timer.setValue(0);
      Animated.loop(rotation.current).start();
    }
  }, [scale, fade, timer]);

  const stopRotation = () => {
    if (rotation.current) {
      rotation.current.stop();
    }
  };

  React.useEffect(() => {
    if (rotation.current === undefined) {
      // Circular rotation in loop
      rotation.current = Animated.timing(timer, {
        duration: DURATION,
        easing: Easing.linear,
        useNativeDriver: true,
        toValue: 1,
        isInteraction: false,
      });
    }

    if (animating) {
      startRotation();
    } else if (hidesWhenStopped) {
      Animated.timing(fade, {
        duration: 200 * scale, // default lightTheme.scale = 1
        toValue: 0,
        useNativeDriver: true,
        isInteraction: false,
      }).start(stopRotation); // TODO: Does this mean to start when stopRotation is.. what?
    }
  }, [animating, fade, hidesWhenStopped, scale, startRotation, timer]);

  const color = indicatorColor || theme.colors?.primary;
  const size =
    typeof indicatorSize === 'string'
      ? DefaultIndicatorSize[indicatorSize as LoaderSize]
      : indicatorSize;

  const frames = (60 * DURATION) / 1000;
  const easing = Easing.bezier(0.4, 0.0, 0.7, 1.0);
  const containerStyle = {
    width: size,
    height: size / 2,
    overflow: 'hidden' as const,
  };

  return (
    <View
      style={[styles.container, style]}
      {...rest}
      accessible
      accessibilityRole="progressbar"
      accessibilityState={{ busy: animating }}
    >
      <Animated.View style={{ width: size, height: size, opacity: fade }} collapsable={false}>
        {[0, 1].map((index) => {
          const inputRange = Array.from(
            new Array(frames),
            (_, frameIndex) => frameIndex / (frames - 1),
          );

          const outputRange = Array.from(new Array(frames), (_, frameIndex) => {
            let progress = (2 * frameIndex) / (frames - 1);
            const rotation = index ? +(360 - 15) : -(180 - 15);

            if (progress > 1.0) {
              progress = 2.0 - progress;
            }

            const direction = index ? -1 : +1;

            return `${direction * (180 - 30) * easing(progress) + rotation}deg`;
          });

          const layerStyle = {
            width: size,
            height: size,
            transform: [
              {
                rotate: timer.interpolate({
                  inputRange: [0, 1],
                  outputRange: [`${0 + 30 + 15}deg`, `${2 * 360 + 30 + 15}deg`],
                }),
              },
            ],
          };

          const viewportStyle = {
            width: size,
            height: size,
            transform: [
              { translateY: index ? -size / 2 : 0 },
              { rotate: timer.interpolate({ inputRange, outputRange }) },
            ],
          };

          const offsetStyle = index ? { top: size / 2 } : null;

          const lineStyle = {
            width: size,
            height: size,
            borderColor: color,
            borderWidth: size / 10,
            borderRadius: size / 2,
          };

          return (
            <Animated.View key={index} style={[styles.layer]}>
              <Animated.View style={layerStyle}>
                <Animated.View style={[containerStyle, offsetStyle]} collapsable={false}>
                  <Animated.View style={viewportStyle}>
                    <Animated.View style={containerStyle} collapsable={false}>
                      <Animated.View style={lineStyle} />
                    </Animated.View>
                  </Animated.View>
                </Animated.View>
              </Animated.View>
            </Animated.View>
          );
        })}
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  layer: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default ActivityIndicator;
