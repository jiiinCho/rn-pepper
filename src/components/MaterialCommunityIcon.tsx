import * as React from 'react';
import { Text, StyleSheet, ViewProps, View } from 'react-native';

import { MD3Colors } from '../styles';

// NOTE: not support for web
// TODO: Investigate more
export const accessibilityProps = {
  accessibilityElementsHidden: true,
  importantForAccessibility: 'no-hide-descendants' as 'no-hide-descendants',
};

export type IconProps = {
  name: string;
  color?: string;
  size: number;
  direction: 'rtl' | 'ltr';
  allowFontScaling?: boolean;
};

let MaterialCommunityIcons: React.ComponentType<
  React.ComponentProps<
    typeof import('react-native-vector-icons/MaterialCommunityIcons').default
  > & {
    color: string;
    pointerEvents: ViewProps['pointerEvents'];
  }
>;

try {
  // Optionally require vector-icons
  MaterialCommunityIcons = require('react-native-vector-icons/MaterialCommunityIcons').default;
} catch (error) {
  let isErrorLogged = false;
  // TODO: Fallback component for icons - add more props if needed
  MaterialCommunityIcons = ({ name, color, size }) => {
    if (!isErrorLogged) {
      if (
        !/(Cannot find module|Module not found|Cannot resolve module)/.test((error as any).message)
      ) {
        console.error(error);
      }

      console.warn(
        `Tried to use the icon '${name}' in a component from 'react-native-paper', but 'react-native-vector-icons/MaterialCommunityIcons' could not be loaded.`,
        `To remove this warning, try installing 'react-native-vector-icons' or use another method to specify icon: https://callstack.github.io/react-native-paper/icons.html.`,
      );

      isErrorLogged = true;
    }

    return (
      // TODO: Add more props if needed
      <View style={[styles.icon]} pointerEvents={'none'}>
        <Text style={{ color, fontSize: size }} selectable={false}>
          â–¡
        </Text>
      </View>
    );
  };
}

const defaultIcon = ({
  name,
  color = MD3Colors.black,
  size,
  direction,
  allowFontScaling,
}: IconProps) => {
  return (
    <MaterialCommunityIcons
      allowFontScaling={allowFontScaling}
      name={name}
      color={color}
      size={size}
      style={[
        {
          transform: [{ scaleX: direction === 'rtl' ? -1 : 1 }],
          lineHeight: size,
        },
        styles.icon,
      ]}
      pointerEvents="none"
      selectable={false}
      {...accessibilityProps}
    />
  );
};

const ICON_BACKGROUND_COLOR = 'transparent';

const styles = StyleSheet.create({
  icon: {
    backgroundColor: ICON_BACKGROUND_COLOR,
  },
});

export default defaultIcon;
