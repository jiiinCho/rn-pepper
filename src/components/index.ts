export { default as Surface } from './Surface';
export type { IconSource } from './Icon';
export { default as ActivityIndicator } from './ActivityIndicator';
export { default as Button } from './Button/Button';
