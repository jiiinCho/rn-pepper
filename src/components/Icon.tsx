import * as React from 'react';
import { I18nManager, Platform, ImageSourcePropType, Image } from 'react-native';

import { useInternalTheme, Consumer as SettingsConsumer } from 'src/core';
import type { ThemeProp } from 'src/types';

import { accessibilityProps } from './MaterialCommunityIcon';

type IconSourceBase = string | ImageSourcePropType;

export type IconSource =
  | IconSourceBase
  | Readonly<{ source: IconSourceBase; direction: 'rtl' | 'ltr' | 'auto' }>
  | ((props: IconProps & { color: string }) => React.ReactNode);

type IconProps = {
  size: number;
  allowFontScaling?: boolean;
};

type Props = IconProps & {
  source: any;
  color?: string;
  direction?: 'rtl' | 'ltr' | 'auto';
  theme?: ThemeProp;
};

const Icon = ({
  source,
  color,
  size,
  theme: themeOverrides,
  direction: customDirection,
  ...rest
}: Props) => {
  const theme = useInternalTheme(themeOverrides);
  const direction =
    customDirection || (typeof source === 'object' && source.direction && source.source)
      ? source.direction === 'auto'
        ? I18nManager.getConstants().isRTL
          ? 'rtl'
          : 'ltr'
        : source.direction
      : null;

  const src =
    typeof source === 'object' && source.direction && source.source ? source.source : source;

  const iconColor = color || (theme.isV3 ? theme.colors.onSurface : theme.colors.text);

  if (isImageSource(src)) {
    return (
      <Image
        {...rest}
        source={src}
        style={[
          { transform: [{ scaleX: direction === 'rtl' ? -1 : 1 }] },
          { width: size, height: size, tintColor: color, resizeMode: `contain` },
        ]}
        {...accessibilityProps}
        accessibilityIgnoresInvertColors
      />
    );
  } else if (typeof src === 'string') {
    return (
      <SettingsConsumer>
        {({ icon }) => {
          return icon({
            name: src,
            color: iconColor,
            size,
            direction,
          });
        }}
      </SettingsConsumer>
    );
  } else if (typeof src === 'function') {
    return src({ color: iconColor, size, direction });
  } else {
    return null;
  }
};

export default Icon;

function isImageSource(source: any) {
  // is source an object with uri?
  return (
    (typeof source === 'object' &&
      source !== null &&
      Object.prototype.hasOwnProperty.call(source, 'uri') &&
      typeof source.uri === 'string') ||
    // source is a module, e.g. require('image')
    typeof source === 'number' ||
    // image url on web
    (Platform.OS === 'web' &&
      typeof source === 'string' &&
      (source.startsWith('data:image') || /\.(bmp|jpg|jpeg|png|gif|svg)$/.test(source)))
  );
}
