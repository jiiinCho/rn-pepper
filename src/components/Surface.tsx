import type * as React from 'react';
import { Animated, Platform, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';

import { useInternalTheme } from 'src/core';
import { isAnimatedValue, overlay, shadow } from 'src/styles';
import type { ThemeProp, MD3EelevationColors } from 'src/types';
import { forwardRef } from 'src/utils';

type Props = React.ComponentPropsWithRef<typeof View> & {
  children: React.ReactNode;
  style?: Animated.WithAnimatedValue<StyleProp<ViewStyle>>;
  /**
   * @supported Available in v5.x with theme v3
   * Changes shadows and background in iOS and Android.
   * Used to create UI hierarchy between components
   * Note: you should use `elevation` property instead of `style={{ elevation: 4 }}`
   */
  elevation?: 0 | 1 | 2 | 3 | 4 | 5 | Animated.Value;
  theme?: ThemeProp;
  testID?: string;
  ref?: React.RefObject<View>;
};

const MD2Surface = forwardRef<View, Props>(
  ({ style, theme: overrideTheme, ...rest }: Omit<Props, 'elevation'>, ref) => {
    const { elevation = 4 } = (StyleSheet.flatten(style) || {}) as ViewStyle;
    const { dark: isDarkTheme, mode, colors } = useInternalTheme(overrideTheme);

    return (
      <Animated.View
        ref={ref}
        {...rest}
        style={[
          {
            backgroundColor:
              isDarkTheme && mode === 'adaptive'
                ? overlay(elevation, colors?.surface)
                : colors?.surface,
          },
          elevation ? shadow(elevation) : null,
          style,
        ]}
      />
    );
  },
);

// const { foo = 'a' } = {};
// console.log(foo); -> a

/**
 * Surface is a basic container that can give depth to an element with elevation shadow.
 * On dark theme with `adaptive` mode, surface is constructed by placing a semi-transparent white overlay over a component surface
 * Overlay and shadow can be applied by specifying the `elevation` property both on Android and iOS.
 *
 * `adaptive` mode is a feature that allows your app to automatically switch between light and dark themes based on the device's system settings.
 * Both iOS and Android have adaptive mode:
 * - On iOS, this feature is called "Automatic" mode (available from iOS 13.):
 *  When the user enables this mode, the device will automatically switch between light and dark modes based on the time of day or the user's location.
 * - On Android, the feature is called "Dark Theme" or "Dark Mode" (available from Android 10):
 *  When the user enables this mode, the device will switch between light and dark modes based on system settings, such as the time of day
 *
 * <div class="screenshots">
 *  <figure>
 *    <img class="medium" src="screenshots/surface-android.png" />
 *    <figcaption>Surface on Android</figcaption>
 *  </figure>
 *  <figure>
 *    <img class="medium" src="screenshots/surface-ios.png" />
 *    <figcaption>Surface on iOS</figcaption>
 *  </figure>
 * </div>
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { Surface, Text } from 'react-native-papper';
 * import { StyleSheet } from 'react-native';
 *
 * const MyComponent = () => (
 *  <Surface style={styles.surface elevation={4}>
 *    <Text>Surface</Text>
 *  </Surface>
 * )
 *
 * export default MyComponent;
 */

const IOSSurfaceOptions = {
  iOSShadowOutputRanges: [
    { shadowOpacity: 0.15, height: [0, 1, 2, 4, 6, 8], shadowRadius: [0, 3, 6, 8, 10, 12] },
    { shadowOpacity: 0.3, height: [0, 1, 1, 1, 2, 4], shadowRadius: [0, 1, 2, 3, 3, 4] },
  ],
  shadowColor: '#000',
  inputRange: [0, 1, 2, 3, 4, 5],
};

const Surface = forwardRef<View, Props>(
  (
    { elevation = 1, children, theme: overridenTheme, style, testID = 'surface', ...props }: Props,
    ref,
  ) => {
    const theme = useInternalTheme(overridenTheme);

    if (!theme.isV3) {
      return (
        <MD2Surface {...props} theme={theme} style={style} ref={ref}>
          {children}
        </MD2Surface>
      );
    }

    const { colors } = theme;
    const inputRange = [0, 1, 2, 3, 4, 5];
    const backgroundColor = (() => {
      if (isAnimatedValue(elevation)) {
        return elevation.interpolate({
          inputRange,
          outputRange: inputRange.map((inputElevation) => {
            const elevationLevel = `level${inputElevation}}` as keyof MD3EelevationColors;
            return colors.elevation?.[elevationLevel];
          }),
        });
      }

      return colors.elevation?.[`level${elevation}`];
    })();

    if (Platform.OS === 'web') {
      return (
        <Animated.View
          {...props}
          ref={ref}
          testID={testID}
          style={[{ backgroundColor }, elevation ? shadow(elevation, theme.isV3) : null, style]}
        >
          {children}
        </Animated.View>
      );
    }

    if (Platform.OS === 'android') {
      const elevationLevel = [0, 3, 6, 9, 12, 15];

      const getElevationAndroid = () => {
        if (isAnimatedValue(elevation)) {
          return elevation.interpolate({
            inputRange,
            outputRange: elevationLevel,
          });
        }

        return elevationLevel[elevation];
      };

      const { margin, padding, transform, borderRadius } = (StyleSheet.flatten(style) ||
        {}) as ViewStyle;

      const outerLayerStyles = { margin, padding, transform, borderRadius };
      const sharedStyle = [{ backgroundColor }, style];

      return (
        <Animated.View
          {...props}
          testID={testID}
          ref={ref}
          style={[
            { backgroundColor, transform },
            outerLayerStyles,
            sharedStyle,
            { elevation: getElevationAndroid() },
          ]}
        >
          {children}
        </Animated.View>
      );
    }

    const { outerLayerViewStyles, innerLayerViewStyles, sharedStyle } = sortIOSStyles({
      style,
      backgroundColor,
    });

    const getIOSShadowStyles = (layer: 0 | 1) => {
      return isAnimatedValue(elevation)
        ? getStyleForAnimatedShadowLayer(layer, elevation)
        : getStyleForShadowLayer(layer, elevation);
    };

    return (
      <Animated.View
        style={[getIOSShadowStyles(0), outerLayerViewStyles]}
        testID={`${testID}-outer-layer`}
      >
        <Animated.View
          style={[getIOSShadowStyles(1), innerLayerViewStyles]}
          testID={`${testID}-inner-layer`}
        >
          <Animated.View {...props} testID={testID} style={sharedStyle}>
            {children}
          </Animated.View>
        </Animated.View>
      </Animated.View>
    );
  },
);

function sortIOSStyles({
  style,
  backgroundColor,
}: {
  style: Props['style'];
  backgroundColor: string | Animated.AnimatedInterpolation<string | number>;
}) {
  const { position, alignSelf, top, left, right, bottom, start, end, flex, ...restStyle } =
    (StyleSheet.flatten(style) || {}) as ViewStyle;

  const absoluteStyles = {
    position,
    alignSelf,
    top,
    right,
    bottom,
    left,
    start,
    end,
  };

  const sharedStyle = [{ backgroundColor, flex }, restStyle];
  const innerLayerViewStyles = [{ flex }];
  const outerLayerViewStyles = [absoluteStyles, innerLayerViewStyles];

  return { outerLayerViewStyles, innerLayerViewStyles, sharedStyle };
}

function getStyleForAnimatedShadowLayer(layer: 0 | 1, elevation: Animated.Value) {
  const { iOSShadowOutputRanges, shadowColor, inputRange } = IOSSurfaceOptions;

  return {
    shadowColor,
    shadowOpacity: elevation.interpolate({
      inputRange: [0, 1],
      outputRange: [0, iOSShadowOutputRanges[layer].shadowOpacity],
      extrapolate: 'clamp',
    }),
    shadowOffset: {
      width: 0,
      height: elevation.interpolate({
        inputRange,
        outputRange: iOSShadowOutputRanges[layer].height,
      }),
    },
    shadowRadius: elevation.interpolate({
      inputRange,
      outputRange: iOSShadowOutputRanges[layer].shadowRadius,
    }),
  };
}

function getStyleForShadowLayer(layer: 0 | 1, elevation: 0 | 1 | 2 | 3 | 4 | 5) {
  const { iOSShadowOutputRanges, shadowColor } = IOSSurfaceOptions;

  return {
    shadowColor,
    shadowOpacity: elevation ? iOSShadowOutputRanges[layer].shadowOpacity : 0,
    shadowOffset: {
      width: 0,
      height: iOSShadowOutputRanges[layer].height[elevation],
    },
    shadowRadius: iOSShadowOutputRanges[layer].shadowRadius[elevation],
  };
}

export default Surface;
