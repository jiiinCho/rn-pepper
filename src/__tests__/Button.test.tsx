import * as React from 'react';
// import { StyleSheet } from 'react-native';

import renderer from 'react-test-renderer';

import { Button } from 'components';

// const styles = StyleSheet.create({
//   flexing: {
//     flexDirection: 'row-reverse',
//   },
// });

it('renders text button by default', () => {
  const tree = renderer.create(<Button label="Text Button" />).toJSON();
  expect(tree).toMatchSnapshot();
});
