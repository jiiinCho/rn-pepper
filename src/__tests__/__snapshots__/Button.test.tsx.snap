// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text button by default 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "start": undefined,
      "top": undefined,
    }
  }
  testID="button-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="button-container-inner-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "flex": undefined,
          "minWidth": 64,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            false,
            {
              "borderRadius": undefined,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 12,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(103, 80, 164, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          >
            Text Button
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
